code/util/fetch_data_docDB.py

```diff
@@ -109,32 +109,32 @@ def strip_dict_for_id(co_asset_id_dict_list):
             
     return result_list                   
 
-def fetch_fip_data(client):
-    # search for records that have the "fib" (for fiber photometry) modality in data_description
-    logger.warning("fetching 'fib' records...")
-    modality_results = client.retrieve_docdb_records(
-        filter_query={"data_description.modality.abbreviation": "fib"},
+def fetch_dynamic_foraging_data(client):
+    # To compare the new FIP pipeline with my temporary pipeline for all dynamic foraging sessions,
+    # let's directly query the software name
+    logger.warning("fetching 'dynamic foraging' in software name...")
+    software_name_results = client.retrieve_docdb_records(
+        filter_query={"session.data_streams.software.name": "dynamic-foraging-task",
+                      "name": {"$not": {"$regex": ".*processed.*"}}, # only raw data
```

Comment:
I trust your code, @hanhou , but i would check with @dyf if this is the correct query.

Reply:
Yeah, I mentioned a caveat about this in the PR description
>Some older sessions may not have correct session.data_streams.software.name (this is not necessarily a problem since our goal is to ensure all uploaded data have the correct fields)

Reply:
fine by me. please include this in a discussion or elsewhere so when folks use streamlit they know that these are known inaccuracies.

Reply:
> fine by me. please include this in a discussion or elsewhere so when folks use streamlit they know that these are known inaccuracies.

created an issue here https://github.com/AllenNeuralDynamics/aind-physio-arch/issues/373